sdk/storage/storage-blob/review/storage-blob.api.md

```diff
@@ -1755,6 +1798,7 @@ export class PageBlobClient extends BlobClient {
     create(size: number, options?: PageBlobCreateOptions): Promise<PageBlobCreateResponse>;
     getPageRanges(offset?: number, count?: number, options?: PageBlobGetPageRangesOptions): Promise<PageBlobGetPageRangesResponse>;
     getPageRangesDiff(offset: number, count: number, prevSnapshot: string, options?: PageBlobGetPageRangesDiffOptions): Promise<PageBlobGetPageRangesDiffResponse>;
+    getPageRangesDiff(offset: number, count: number, prevSnapshotUrl: string, options?: PageBlobGetPageRangesDiffOptions): Promise<PageBlobGetPageRangesDiffResponse>;
```

Comment:
I think an overload is maybe overkill here? The two signatures are identical except for the parameter name. If we get into a world where the return types are different, users will have no way to select the second signature and will always get types on the first. In general we should only have overloads when we need to document two distinct usages of the API, and I'm not sure that motivation applies here.

Thoughts on having one signature with prevSnapshot being documented as either a url or a datetime value?

Reply:
Thanks.
Since the url option is only available for managed disk accounts, is it better if we add a new API like `GetManagedDiskPageRangesDiff`? 
 .NET SDK chose this way: https://github.com/Azure/azure-sdk-for-net/pull/9195/files#diff-05327b3d0b26fefc045c8c30eda382d1R1054

If this URL option will be limited to managed disk account in the future, I think this would be a good choice.

Reply:
I think it is a good idea to use one signature. Python may be doing this.

Reply:
@ljian3377 I think I agree with @jeremymeng for the most part. I've also chatted with @annatisch and @tg-msft a bit and I think we should take the approach that adds the snapshot URL in the most minimal-impact way. I feel that not doing overloads and just taking a string (and documenting that you have to pass a URL for managed disks) is the best approach for now.

Reply:
I agree with @bterlson, except that I would go further and not even document the behaviour - at least not for this release.
If we do then we have to support that behaviour in the public API indefinitely - which seems risky given the niche nature of this feature.

Reply:
I am also ok with not documenting this behavior given how niche-y it is. @tg-msft how do you feel?

Reply:
I'm fine with that so long as folks on the Storage team are okay with it too.

Reply:
I am fine with one signature and have doubts about the 'not even document' part as we have customers from Dell asking for this feature.